polkadot-parachains/src/rpc.rs

```diff
@@ -20,14 +20,24 @@
 
 use std::sync::Arc;
 
-use polkadot_primitives::v1::{AccountId, Balance, Block, Nonce};
 use sc_client_api::AuxStore;
 pub use sc_rpc::{DenyUnsafe, SubscriptionTaskExecutor};
 use sp_api::ProvideRuntimeApi;
 use sp_block_builder::BlockBuilder;
 use sp_blockchain::{Error as BlockChainError, HeaderBackend, HeaderMetadata};
 use txpool_api::TransactionPool;
 
+/// Alias to 512-bit hash when used in the context of a transaction signature on the chain.
+pub type Signature = sp_runtime::MultiSignature;
+/// Some way of identifying an account on the chain. We intentionally make it equivalent
+/// to the public key of our transaction signing scheme.
+pub type AccountId = <<Signature as sp_runtime::traits::Verify>::Signer as sp_runtime::traits::IdentifyAccount>::AccountId;
+type BlockNumber = u32;
+type Header = sp_runtime::generic::Header<BlockNumber, sp_runtime::traits::BlakeTwo256>;
+pub type Block = sp_runtime::generic::Block<Header, sp_runtime::OpaqueExtrinsic>;
+type Balance = u128;
+type Nonce = u32;
```

Comment:
i dont think this is right either. shouldnt be that we are redefining all these types here.

should make the function generic over these types, or properly import them

Reply:
Moved them to primitives module and imported the common primitives in rpc.rs and service.rs.